source/docs/software/advanced-programming/trajectorygeneration.rst

```diff
@@ -0,0 +1,157 @@
+* ``endVelocity``: This is the velocity at which the trajectory will end. This should be zero if you want the robot to stop completely at the end of the trajectory.
+* ``maxVelocity``: This is the maximum velocity that the robot should reach when traversing the trajectory.
+* ``maxAcceleration``: This is the maximum acceleration that the robot should experience when traversing the trajectory.
+* ``reversed``: This is whether the robot should travel backwards or not.
+
+Note: The ``reversed`` property simply represents whether the robot is traveling backward. If you specify four waypoints, a, b, c, and d, the robot will still travel in the same order through the waypoints when the ``reversed`` flag is set to ``true``. This also means that you must account for the direction of the robot when providing the waypoints. For example, if your robot is facing your alliance station wall and travels backwards to some field element, the starting waypoint should have a rotation of 180 degrees.
+
+Here is an example of generating a trajectory using clamped cubic splines for the 2018 game, FIRST Power Up:
+
+.. tabs::
+
+   .. code-tab:: java
+
+      final double startVelocity = 0;
+      final double endVelocity = 0;
+      final double maxVelocity = feetToMeters(12.0);
+      final double maxAccel = feetToMeters(12);
+
+      // 2018 cross scale auto waypoints.
+      var sideStart = new Pose2d(feetToMeters(1.54), feetToMeters(23.23),
+          Rotation2d.fromDegrees(-180));
+      var crossScale = new Pose2d(feetToMeters(23.7), feetToMeters(6.8),
+          Rotation2d.fromDegrees(-160));
+
+      var interiorWaypoints = new ArrayList<Translation2d>();
+      waypoints.add(new Translation2d(feetToMeters(14.54), feetToMeters(23.23));
+      waypoints.add(new Translation2d(feetToMeters(21.04), feetToMeters(18.23));
+
+      var trajectory =  TrajectoryGenerator.generateTrajectory(
+          sideStart,
+          interiorWaypoints,
+          crossScale,
+          new ArrayList<>(),
+          startVelocity,
+          endVelocity,
+          maxVelocity,
+          maxAccel,
+          true);
+
+   .. code-tab:: c++
+
+      const units::meters_per_second_t startVelocity = 0_mps;
+      const units::meters_per_second_t endVelocity = 0_mps;
+      const units::meters_per_second_t maxVelocity = 12_fps;
+      const units::meters_per_second_squared_t maxAcceleration = 12_fps_sq;
+
+      // 2018 cross scale auto waypoints
+      const Pose2d sideStart{1.54_ft, 23.23_ft, Rotation2d(180_deg)};
+      const Pose2d crossScale{23.7_ft, 6.8_ft, Rotation2d(-160_deg)};
+
+      auto interiorWaypoints = std::vector<Translation2d>{
+        frc::Translation2d{14.54_ft, 23.23_ft},
+        frc::Translation2d{21.04_ft, 18.23_ft}};
+
+      auto trajectory = TrajectoryGenerator::GenerateTrajectory(
+        sideStart, interiorWaypoints, crossScale, std::vector<std::unique_ptr<TrajectoryConstraint>>(), startVelocity,
```

Comment:
You can do `{}` instead of `std::vector<std::unique_ptr<TrajectoryConstraint>>()`.